docs/os-ref.md

```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
 
-#### 5.2.1 Script executors and EVMScriptRegistry
+Protecting an action behind the ACL is done in the smart contract by simply adding the authentication modifiers `auth` or `authP()` to the action. On executing the action, the `auth` or `authP` modifier checks with the Kernel whether the entity performing the call holds the required role or not.
 
-EVMScriptExecutors must follow this interface:
+`auth` is capable of defining a *binary* permission—either yes or no:
 
 ```solidity
+function customAction() auth(CUSTOM_ACTION_ROLE) {
+}
+```
 
-interface IEVMScriptExecutor {
-    function execScript(bytes script, bytes input, address[] blacklist) external returns (bytes);
+`authP` allows you to pass a number of parameters, that can then be used in the [ACL's parameterization](#parameter-interpretation) for each permission. This allows you to define powerful permissions with highly granular controls based on the inputs of an action:
+
+```solidity
+bytes32 public TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
+function transfer(uint256 amount) authP(TRANSFER_ROLE, arr(amount)) {
 }
 ```
 
-Because script executors get are called with a `delegatecall`, in order to prevent
-self-destructs, `IEVMScriptExecutor.execScript(...)` MUST return
-at least 32 bytes so in case an executor `selfdestruct`s it could be detected.
+`authP` takes a `uint256[]` as its second argument, and a number of `arr()` syntatical sugar helpers are exposed by default by AragonApp to help construct this array when using different argument types.
 
-##### 5.2.1.1 CallsScript
-A simple way to concatenate multiple calls. It cancels the operation if any of the calls fail.
+Finally, AragonApp also exposes a public getter for checking if an entity can perform a certain action:
 
-- **Script body:** (See source code file for spec of the payload)
-- **Input:** None
-- **Output:** None.
-- **Blacklist:** Entire script reverts if a call to one of the addresses in the blacklist is performed.
+```solidity
+function canPerform(address sender, bytes32 role, uint256[] params) public view returns (bool);
+```
 
-##### 5.2.1.1 DelegateScript
-`delegatecalls` into a given contract, which basically allows for any arbitrary computation within the EVM in the caller’s context.
+> **Note**
+>
+> Apps have the choice of which actions to protect behind the ACL, as some actions may make sense to be completely public. However, any publicly exposed state-changing function should *most likely* be protected.
 
-- **Script body:** Address of the contract to make the call to.
-- **Input:** `calldata` for the `delegatecall` that will be performed.
-- **Output:** raw return data of the call.
-- **Blacklist:** impossible to enforce. If there are any addresses in the blacklist the script will revert as it is not possible to check whether a particular address will be called.
+**Lifecycle of an AragonApp call requiring the ACL**
 
-##### 5.2.1.3 DeployDelegateScript
+![](/docs/assets/os-app-call.gif)
```

Comment:
could we increase the font sizes on this diagram too?

Reply:
maybe I could work on all the diagrams as a batch?  Do you have the source files for all of them and what application do you use to edit?

Reply:
I'm not sure we still have these somewhere, especially since this one was animated... @izqui made them originally and might.